src/bin/pgcopydb/table-data.c

```diff
@@ -1196,27 +1268,28 @@ copydb_table_create_lockfile(CopyDataSpec *specs,
 
 
 /*
- * copydb_mark_table_as_done creates the table doneFile with the expected
- * summary content. To create a doneFile we must acquire the synchronisation
- * semaphore first. The lockFile is also removed here.
+ * copydb_save_copy_progress saves the duration and the number of
+ * bytes transmitted about the copy progress into the catalog.
  */
 bool
-copydb_mark_table_as_done(CopyDataSpec *specs,
-						  CopyTableDataSpec *tableSpecs)
+copydb_save_copy_progress(CopyDataSpec *specs,
+						  uint64_t count,
+						  uint64_t bytes,
+						  instr_time lastSavingTimeMs)
 {
 	DatabaseCatalog *sourceDB = &(specs->catalogs.source);
 
-	if (!summary_finish_table(sourceDB, tableSpecs))
-	{
-		/* errors have already been logged */
-		return false;
-	}
```

Comment:
I still expected to see a new function for handling updates during the COPY run, and to keep the existing function for marking the operation as done by updating the `done_time_epoch`. As I don't see any changes to the `done_time_epoch` handling (other than lacking an update), can you explain what's happening here?

Reply:
Hint: see `summary_table_count_parts_done()`, or `copydb_objectid_has_been_processed_already`, or `copydb_objectid_has_been_processed_already()`, where we use the `done_time_epoch` field.

Reply:
Basically, we don't call `summary_finish_table` during the copy operation, just at the end as before.